nomenclature/codelist.py

```diff
@@ -364,6 +335,35 @@ def read_excel(cls, name, source, sheet_name, col, attrs=None):
 
         return cls(name=name, mapping=mapp)
 
+    def check_illegal_characters(
+        self, config: NomenclatureConfig = None
+    ) -> Dict[str, Code]:
+        """Check that no illegal characters are left in codes after tag replacement"""
+        forbidden = ["{", "}"]
+        if config and config.illegal_characters:
+            forbidden += config.illegal_characters
+
+        def _check_string(value):
+            if isinstance(value, str):
+                if any(char in value for char in forbidden):
+                    raise ValueError(
```

Comment:
If you have a number of illegal errors, this can become extremely frustrating to fix because you get the errors one by one.
You can use the `ErrorCollector` class for this. There are a bunch of examples in nomenclature on how to use it.

Reply:
Similar to my previous comment, you marked this as resolved but did not implement it or provide a reason why you did not implement it.